src/lsp-server.ts

```diff
@@ -1077,6 +1084,32 @@ export class LspServer {
         return applied;
     }
 
+    async willRenameFiles(params: lsp.RenameFilesParams, token?: lsp.CancellationToken): Promise<lsp.WorkspaceEdit> {
+        const toTextEdit = (codeEdit: ts.server.protocol.CodeEdit): lsp.TextEdit => {
+            return {
+                range: {
+                    start: Position.fromLocation(codeEdit.start),
+                    end: Position.fromLocation(codeEdit.end),
+                },
+                newText: codeEdit.newText,
+            };
+        };
+
+        const workspaceEdit: lsp.WorkspaceEdit = { changes: {} };
+        for (const rename of params.files) {
+            const codeEdits = await this.getEditsForFileRename(rename.oldUri, rename.newUri, token);
+            for (const codeEdit of codeEdits) {
+                const uri = pathToUri(codeEdit.fileName, this.documents);
+                if (!workspaceEdit.changes) {
+                    workspaceEdit.changes = {};
+                }
+                const textEdits = workspaceEdit.changes[uri] || (workspaceEdit.changes[uri] = []);
+                textEdits.push(...codeEdit.textChanges.map(toTextEdit));
+            }
+        }
+        return workspaceEdit;
```

Comment:
Just a style thing but this is shorter and cleaner:

```suggestion
        const changes: lsp.WorkspaceEdit['changes'] = {};
        for (const rename of params.files) {
            const codeEdits = await this.getEditsForFileRename(rename.oldUri, rename.newUri, token);
            for (const codeEdit of codeEdits) {
                const uri = pathToUri(codeEdit.fileName, this.documents);
                const textEdits = changes[uri] || (changes[uri] = []);
                textEdits.push(...codeEdit.textChanges.map(toTextEdit));
            }
        }
        return { changes };
```

Reply:
I looked at `rename` function. Will fix there also.